Build fails with the following issue

Run npm run build

> michelin-star-hunter-frontend@1.0.0 build
> tsc && vite build

Error: src/pages/DashboardPage.tsx(291,39): error TS18048: 'visit.moodRating' is possibly 'undefined'.
Error: src/pages/ProfilePage.tsx(160,35): error TS18048: 'visit.moodRating' is possibly 'undefined'.
Error: src/pages/RestaurantDetailPage.tsx(150,7): error TS2353: Object literal may only specify known properties, and 'bestDish' does not exist in type '{ restaurantId: string; dateVisited: string; notes?: string | undefined; }'.
npm error Lifecycle script `build` failed with error:
npm error code 2
npm error path /home/runner/work/MichelinCompare/MichelinCompare/frontend
npm error workspace michelin-star-hunter-frontend@1.0.0
npm error location /home/runner/work/MichelinCompare/MichelinCompare/frontend
npm error command failed
npm error command sh -c tsc && vite build
Error: Process completed with exit code 2.

@claude can you fix this please.

--- a/frontend/src/pages/DashboardPage.tsx
+++ b/frontend/src/pages/DashboardPage.tsx
@@ -288,7 +288,7 @@ const DashboardPage = () => {
                               <Star
                                 key={i}
                                 className={`h-3 w-3 ${
-                                  i < visit.moodRating ? 'text-yellow-400 fill-current' : 'text-gray-300'
+                                  i < (visit.moodRating ?? 0) ? 'text-yellow-400 fill-current' : 'text-gray-300'
                                 }`}
                               />
                             ))}

--- a/frontend/src/pages/ProfilePage.tsx
+++ b/frontend/src/pages/ProfilePage.tsx
@@ -157,7 +157,7 @@ const ProfilePage = () => {
                           <Star
                             key={i}
                             className={`h-3 w-3 ${
-                              i < visit.moodRating ? 'text-yellow-400 fill-current' : 'text-gray-300'
+                              i < (visit.moodRating ?? 0) ? 'text-yellow-400 fill-current' : 'text-gray-300'
                             }`}
                           />
                         ))}

--- a/frontend/src/services/api.ts
+++ b/frontend/src/services/api.ts
@@ -62,7 +62,7 @@ export const restaurantAPI = {
 };
 
 export const visitAPI = {
-  createVisit: (data: { restaurantId: string; dateVisited: string; notes?: string }) =>
+  createVisit: (data: { restaurantId: string; dateVisited: string; notes?: string; bestDish?: string; occasion?: string; moodRating?: number }) =>
     api.post<{ message: string; visit: UserVisit; pointsEarned: number }>('/visits', data),
 
   getUserVisits: (page = 1, limit = 20) =>

